server/v2/cometbft/abci.go

```diff
@@ -259,17 +265,42 @@ func (c *Consensus[T]) InitChain(ctx context.Context, req *abci.RequestInitChain
 		AppHash: nil,
 	}
 
-	genesisState, err := c.app.InitGenesis(ctx, genesisHeaderInfo, consMessages, req.AppStateBytes)
+	genesisState, err := c.app.InitGenesis(
+		ctx,
+		genesisHeaderInfo,
+		consMessages,
+		req.AppStateBytes,
+		c.txCodec)
 	if err != nil {
 		return nil, fmt.Errorf("genesis state init failure: %w", err)
 	}
 
+	// TODO: this query forms a dependency on x/staking, remove it.
+	// we have agreed the code smell is the tight coupling of validators and x/staking
+	// future work will be to query validators from state using only a abstraction in core
+	res, err := c.app.QueryWithState(ctx, genesisState, &staking.QueryValidatorsRequest{})
```

Comment:
I think this query can be removed since a genesis block is being executed which should update validators in EndBlock, however I can't test the flow yet.. blocked on https://github.com/cosmos/cosmos-sdk/pull/19949